frontend/src/shared/reducers.js

```diff
@@ -33,6 +33,21 @@ const auth = (state = {}, action) => {
   }
 }
 
+const performingRole = (state = {type: 'ADMIN'}, action) => {
+  let newState = state
+  switch (action.type) {
+    case actions.JUDGE_ROLE:
+      newState.type = 'JUDGE'
+      return newState
+    case actions.ADMIN_ROLE:
+      newState.type = 'ADMIN'
+      return newState  
+    default:
+      return state
+  }
+}
+
 export default combineReducers({
-  auth
+  auth,
+  performingRole
```

Comment:
this smells a bit odd to me... I feel like `performingRole` should be a a key in the `auth` reducer that maps to a string, such that auth looks like:

```json
{
   "token": "...",
   "user": {
     "username": "souperManne",
     "firstName": "Clark",
     "lastName": "Kent",
     "type": "ADMIN",
     "createdAt": "...",
     "iat": 1516390788
  },
  "performingRole": "JUDGE"
}
```

... which I think might be a bit cleaner for everything, especially considering the state currently doesn't clear when you logout (bug or feature tho?)

Reply:
I think moving it to auth could work. This task was mostly me learning to use redux and I just wasn't thinking of designing it that way when I started.